### PR TITLE
refactor(core): Centralize executions retrieval (no-changelog)

### PR DESCRIPTION
**Pull request by ivov**
This PR implements the BE changes for Alex's new executions list, plus some additional work:
- Refactor parts of the executions controller, service, repository, and middleware.
- Expand test coverage for executions controller, service, repository, and middleware.
- Adjust FE stores to use new endpoints.

Even with the prior three PRs #8349, #8358, #8371 and the additional work here, executions are weighed down by a lot more tech debt, which it'd be best to address in future, to control the size of this PR.

### CODE REVIEW

**Comment by ivov** [comment]
I merged `master` into Alex's branch and his into mine today, which started causing two unit tests for the executions list UI to fail, which I assume is expected as he is midway working on it and currently on vacation. From basic testing, the executions list UI works fine with my endpoint adjustments, so I take it those two unit tests should not block merging into his branch.

File: packages/cli/src/executions/execution.types.ts
Comment (krynble) [inline packages/cli/src/executions/execution.types.ts:44]: I guess this means we're changing the contract heavily right? Just making sure that Alex is onboard with it.

**Comment by ivov** [comment]
Self-reminder to add tests for https://github.com/n8n-io/n8n/pull/8480

File: packages/cli/src/executions/parse-range-query.middleware.ts
Comment (krynble) [inline packages/cli/src/executions/parse-range-query.middleware.ts:28]: We might want to put an upper bound to `range` wdyt? I know it's about to fill the gaps but we need to prevent the n8n backend from crashing because a bug in the FE requested too much data.

What about limiting it to 100?

File: packages/editor-ui/src/api/workflows.ts
Comment (krynble) [inline packages/editor-ui/src/api/workflows.ts:31]: This will be removed right? This is just here to prevent breaking things entirely I guess

File: packages/cli/src/executions/execution.types.ts
Comment (ivov) [reply packages/cli/src/executions/execution.types.ts:44]: This type is the result of the parsing done by the middleware. I tested with the UI and it works. From what I can tell this does not change the contract.

File: packages/editor-ui/src/api/workflows.ts
Comment (ivov) [reply packages/editor-ui/src/api/workflows.ts:31]: The old endpoint does not exist anymore, so to test on the UI I had to update it, but Alex has the final say on this.

File: packages/cli/src/databases/repositories/execution.repository.ts
Comment (krynble) [reply packages/cli/src/databases/repositories/execution.repository.ts:613]: Also let's make sure this array is `ExecutionStatus[]` so we can track `failed` once we decide to remove it

**Review by krynble — APPROVED** [review]

**Pull request merged**
